mcs/class/System.Security/ReferenceSources/DataProtector.cs

```diff
@@ -0,0 +1,224 @@
+//------------------------------------------------------------
+// Copyright (c) Microsoft Corporation.  All rights reserved.
+//------------------------------------------------------------
+
+using System.Collections.Generic;
+using System.IO;
+using System.Runtime.CompilerServices;
+using System.Text;
+using Crypto = System.Security.Cryptography.Translation;
+
+namespace System.Security.Cryptography
+{
+    // Data protectors should be derived from this class
+    public abstract class DataProtector
```

Comment:
Where did you get this file from?

Reply:
https://referencesource.microsoft.com/#System.Security/system/security/cryptography/dataprotector.cs,07e03c337e51f308